File: DSA/SparshJain/Day_1_Question_1.cpp
#include<iostream>
using namespace std;
int Possibilities(int arr[],int size, int Birth_Date , int Birth_Month ){
   
    

int ways=0;    
    for(int i=0;i<=size-Birth_Month;i++){
        int sum=0;
        
        for(int j=i;j<i+Birth_Month;j++){
        
            sum+=arr[j];
            }
        
        if(sum==Birth_Date){
            ways++;
        }
    }
    return ways;

}
int main(){
    int chocoBar[5]={2, 2, 1, 3, 2};
    int Birth_Date;
    int Birth_Month;
    cout<<"Enter the Birth date : \n";
    cin>>Birth_Date;
    cout<<"Enter the Birth month : \n";
    cin>>Birth_Month;
    int count=Possibilities(chocoBar,5,Birth_Date,Birth_Month);
    cout<<count;

}

File: DSA/Aaditya_Vijayvargiya/Day1Question1.java
// Day1Question1Raju have a candy where each square has a number. Rohan wants to find out how many ways she can break the bar into a contiguous segment such that:

// 1. The segment length matches Raju's birth month.
// 2. The sum of the numbers in the segment equals Raju's birth day.


    public static int segment(int[]arr, int month, int day){
        int segmentSize=month;
        int i,j;
        int sum=0;
        int count=0;
        for(i=0;i<=arr.length-segmentSize;i++){
            sum=0;
            for(j=i;j<i+segmentSize;j++){
                sum+=arr[j];
            }
            if(sum==day){
                count+=1;
                
            }
        }
        return count;
    }
